zenCssButton

brittainmark opened the pull request:
fixes  #5545

scottcwilson commented:
Product 108 in the test dataset (a sold out product).  

Before: 
<img width="451" alt="before" src="https://github.com/zencart/zencart/assets/4391638/edc4d2fb-b5d9-4780-9739-d0a0656169cf">

After: 

<img width="434" alt="after" src="https://github.com/zencart/zencart/assets/4391638/2f787b66-f536-4681-98e2-762e96cc48c4">

scottcwilson commented:
Looks right to me.

torvista commented:
Should there not be some visual difference between Ask a Question (link) and Sold Out (information/warning)?

brittainmark commented:
Can add CSS. What do you suggest, red background white text?

scottcwilson commented:
Fair enough, just make the button background color red.

brittainmark commented:
Added Id and css. I have changed the background so it stays the same colour on hover as it did not look right changing colour with no action.
Responsive Classic Template
![image](https://github.com/zencart/zencart/assets/63110817/5bc52884-672c-4c0e-9b6c-35a0b6688cd4)

Classic Contemporary Green
![image](https://github.com/zencart/zencart/assets/63110817/e5d9af58-4c80-45a8-bab2-c132ec641a9d)

scottcwilson commented:
Love it!  Thank you.

lat9 commented on includes/functions/functions_general.php:
```diff
@@ -208,8 +208,13 @@ function zen_get_buy_now_button($product_id, string $buy_now_link, $additional_l
             $alt = BUTTON_SOLD_OUT_ALT; 
-            $return_button = '<span class="text-center">' . zen_image($template->get_template_dir($image, DIR_WS_TEMPLATE, $current_page_base, 'buttons/' . $_SESSION['language'] . '/') . $image, $alt, '', '', '') . '</span>'; 
+            if (strtolower(IMAGE_USE_CSS_BUTTONS) == 'yes') {
+                $return_button =  zenCssButton($image, $alt, 'button', '', 'id=soldOutButton');
```
An `id=` is going to generate invalid HTML if multiple products are 'sold out' on a products' listing page.  Suggest using `class=` instead.

brittainmark replied:
Good spot thanks.
Cannot use class as html_output line 451 will create a duplicate class.
` $css_button = '<input class="' . $mouse_out_class . '" ' . $css_button_js . ' type="submit" value="' . $text . '"' . $tooltip . $parameters . '>';`

I suggest the i remove the parameter. 
The button is identified in css by `span.button_sold_out`

lat9 replied:
Hold off; this should be using
```php
$return_button = zen_image_button($image, $alt);
```
That will generate a `button` with a class of `button_sold_out`.

lat9 replied:
... and the comparison should be strict, i.e. `=== 'yes'`.

brittainmark replied:
I tried zen_image_button that creates a clickable button that takes you to the cart. 
That was why I used zenCssButton. 
ZenCssButton also generate a button with class button_sold_out. Clicking it has no action, the same as the image

lat9 replied:
I'm not seeing that generation of a 'clickable' button.  When I make the change that I suggested (less the CSS updates), I'm presented with an unclickable `span` that says "Sold Out"; here's the hover image
![image](https://github.com/zencart/zencart/assets/2685585/fcdf115c-fd29-428e-a444-a999f3184b1e)

brittainmark replied:
OK I'll try it again. That is where I started. Ignore last push. I'll check it out and push again if ok.

brittainmark replied:
Why should I doubt the master. Works a treat!

lat9 reviewed (changes requested):
Change `id=` to `class=` for valid HTML with multiple sold-out products on a listing.
